### PR TITLE
add wslFeatureInstalled context key

### PR DESCRIPTION
**Pull request by aeschli**
adds a context variable 'wslFeatureInstalled' that is set when the WSL feature is installed.

Once detected, this is remembered in the storage, so the test is no longer made.

### CODE REVIEW

**Comment by aeschli** [comment]
@jrieken Please have a look.
I opted to the built-in context key, as I need it not only for the remote explorer but also for the commands contributed to the remote menu.

File: src/vs/workbench/contrib/remote/electron-sandbox/remote.contribution.ts [hunk 1]
```diff
@@ -131,11 +134,46 @@ class RemoteEmptyWorkbenchPresentation extends Disposable implements IWorkbenchC
...
+		const contextKeyId = 'wslFeatureInstalled';
+		const storageKey = 'remote.wslFeatureInstalled';
+
+		const defaultValue = storageService.getBoolean(storageKey, StorageScope.APPLICATION, undefined);
+
+		const hasWSLFeatureContext = new RawContextKey<boolean>(contextKeyId, !!defaultValue, nls.localize('wslFeatureInstalled', "Whether the platform has the WSL feature installed"));
+		const contextKey = hasWSLFeatureContext.bindTo(contextKeyService);
+
+		if (defaultValue === undefined) {
+			nativeHostService.hasWSLFeatureInstalled().then(res => {
```
Comment (jrieken) [inline src/vs/workbench/contrib/remote/electron-sandbox/remote.contribution.ts:158]: Optionally wait for LifecyclePhase#Eventually when no cached value exists

**Review by jrieken — DISMISSED** [review]

**Comment by aeschli** [comment]
@jrieken I made the change as suggest, please review...

**Review by dbaeumer — APPROVED** [review]
Reviewed that correct life cycle phase is used.

**Comment by aeschli** [comment]
@Tyriar @RMacfarlane This could also be of interest to terminal: if you what to enable/disable functionality based on whether is WSL is/was installed: The `wslFeatureInstalled` context variable is for that. Right now, once we detect WSL, we remember that, and don't check anymore. The idea is that installing WSL is probably quite rare. If the user uninstalls WSL he would still see the commands/views, but on activation we would suggest to install WSL. We'll see how that works for users.

I've seen that the terminal code contains WSL specific code at a few places. The new module src/vs/platform/remote/node/wsl.ts would be a good place to put that code.

**Comment by Tyriar** [comment]
@aeschli thanks for letting us know, I think this is the only place where we could potentially use this which is in the pty host, not the workbench:

https://github.com/microsoft/vscode/blob/0c1f65f9e27ec07b20d3f4fe50d3b6d4861e86e0/src/vs/platform/terminal/node/terminalProfiles.ts#L127-L132

This is done whenever we refresh terminal profiles so the caching wouldn't be needed on our end anyway.

cc @meganrogge

**Pull request merged**